### PR TITLE
docker: remove nsswitch.conf

### PR DESCRIPTION
**Pull request by iavael**
Since docker images are not alpine-based anymore and do support nss, nsswitch.conf hack is not required anymore

### CODE REVIEW

**Review by ahrtr — APPROVED** [review]
Good catch! 

Thank you @iavael 

Could you backport this PR to 3.5 and 3.4?

**Comment by serathius** [comment]
Simplifying dockerfiles has been on my mind for long time. Thanks for looking into this!

**Comment by serathius** [comment]
@ahrtr There is no need to backport this. This is more of a cleanup. Let's not backport cleanups.

**Comment by ahrtr** [comment]
> @ahrtr There is no need to backport this. This is more of a cleanup. Let's not backport cleanups. I doesn't benefit older releases and would only confuse users reading changelog.

Actually it's a minor issue (introduced in one of my previous PRs) to me. There are two duplicate keys "hosts", see line 21 and line 12. 
https://github.com/etcd-io/etcd/blob/215b53cf3b48ee761f4c40908b3874b2e5e95e9f/nsswitch.conf#L21
https://github.com/etcd-io/etcd/blob/215b53cf3b48ee761f4c40908b3874b2e5e95e9f/nsswitch.conf#L12

I suggest to backport the PR.

**Comment by ahrtr** [comment]
Since 3.4.24 isn't released yet, and so no need to update the changelog for 3.4. Note all dockerfile/nsswitch.conf changes are only included in 3.4.24.

**Comment by iavael** [comment]
@ahrtr 
> Could you backport this PR to 3.5 and 3.4?

PRs created

**Comment by ahrtr** [comment]
> @ahrtr
> 
> > Could you backport this PR to 3.5 and 3.4?
> 
> PRs created

Thank you @iavael .  I see that you marked both PRs as draft, anything else you want to update on them? Both of them seems good to me. Please mark them as ready for review if you think they are ready.

**Comment by ahrtr** [comment]
Please also add a changelog item for 3.5 in [CHANGELOG-3.5.md](https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md).

Since the PR adding `nsswitch.conf` isn't included in the latest 3.4 release (3.4.23) yet, so no need to add changelog item for 3.4.

File: CHANGELOG/CHANGELOG-3.5.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ Previous change logs can be found at [CHANGELOG-3.4](https://github.com/etcd-io/
 - Require [Go 1.17+](https://github.com/etcd-io/etcd/pull/15019).
 - Compile with [Go 1.17+](https://go.dev/doc/devel/release#go1.17)
 
+### Other
+- Update [Remove nsswitch.conf from docker image](https://github.com/etcd-io/etcd/pull/15155)
```
Comment (ahrtr) [inline CHANGELOG/CHANGELOG-3.5.md:29]:
```suggestion
- [Remove nsswitch.conf from docker image](https://github.com/etcd-io/etcd/pull/15155)
```

**Review by ahrtr — CHANGES_REQUESTED** [review]
Please rebase this PR and resolve the comment for changelog

File: CHANGELOG/CHANGELOG-3.5.md [hunk 2]
```diff
@@ -4,6 +4,13 @@ Previous change logs can be found at [CHANGELOG-3.4](https://github.com/etcd-io/
 
 <hr>
 
+## v3.5.8 (TBD)
+
+### Other
+- [Remove nsswitch.conf from docker image](https://github.com/etcd-io/etcd/pull/15155)
```
Comment (ahrtr) [inline CHANGELOG/CHANGELOG-3.5.md:10]:
```suggestion
- [Remove nsswitch.conf from docker image](https://github.com/etcd-io/etcd/pull/15161)
```

**Comment by ahrtr** [comment]
Please squash the commits into one.

**Comment by ahrtr** [comment]
@iavael please resolve the conflict and squash the commits.

**Comment by ahrtr** [comment]
There is still conflict.

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @iavael

**Pull request merged**